subgraphs/_reference_/src/sdk/protocols/perpfutures/account.ts

```diff
@@ -0,0 +1,554 @@
+    collateralIn.blockNumber = this.event.block.number;
+    collateralIn.timestamp = this.event.block.timestamp;
+    collateralIn.inputTokens = pool.getInputTokens();
+    collateralIn.outputToken = pool.getOutputToken();
+    collateralIn.inputTokenAmounts = amounts;
+    collateralIn.outputTokenAmount = sharesMinted;
+    collateralIn.amountUSD = this.getAmountUSD(pool.getInputTokens(), amounts);
+    collateralIn.pool = pool.getBytesID();
+    collateralIn.save();
+
+    if (updateMetrics)
+      this.protocol.addTransaction(TransactionType.COLLATERAL_IN);
+    return collateralIn;
+  }
+
+  /**
+   *
+   * @param pool The pool where the collateral was withdrawn.
+   * @param position The position this transaction belongs to as relates to Long or Short but not LP.
+   * @param amounts The amount withdrawn of inputTokens.
+   * @param sharesMinted The amount of shares burnt of outputToken.
+   * @param updateMetrics Optional, defaults to true. If true it will update the pool and protocol TVL and inputTokenBalance.
+   * @returns CollateralOut
+   */
+  collateralOut(
+    pool: Pool,
+    position: Bytes,
+    amounts: BigInt[],
+    sharesBurnt: BigInt,
+    updateMetrics: bool = true
+  ): CollateralOut {
+    const collateralId = this.getIdFromEvent(EventType.WITHDRAW);
+    const collateralOut = new CollateralOut(collateralId);
+
+    collateralOut.hash = this.event.transaction.hash;
+    collateralOut.logIndex = this.event.logIndex.toI32();
+    collateralOut.protocol = this.protocol.getBytesID();
+    collateralOut.position = position;
+    collateralOut.to = this.account.id;
+    collateralOut.from = pool.getBytesID();
+    collateralOut.account = this.account.id;
+    collateralOut.blockNumber = this.event.block.number;
+    collateralOut.timestamp = this.event.block.timestamp;
+    collateralOut.inputTokens = pool.getInputTokens();
+    collateralOut.outputToken = pool.getOutputToken();
+    collateralOut.inputTokenAmounts = amounts;
+    collateralOut.outputTokenAmount = sharesBurnt;
+    collateralOut.amountUSD = this.getAmountUSD(pool.getInputTokens(), amounts);
+    collateralOut.pool = pool.getBytesID();
+    collateralOut.save();
+
+    if (updateMetrics)
+      this.protocol.addTransaction(TransactionType.COLLATERAL_OUT);
+    return collateralOut;
+  }
+
+  /**
+   *
+   * @param pool The pool where the liquidation happened.
+   * @param asset Asset repaid (borrowed)
+   * @param collateralToken Token which was the collateral
+   * @param amountLiquidated Amount of collateral liquidated in native units
+   * @param liquidator Account that carried out the liquidation
+   * @param liquidatee Account that got liquidated
+   * @param position The position this Liquidate belongs to
+   * @param profitUSD Amount of profit from liquidation in USD
+   * @param updateMetrics Optional, defaults to true. If true it will update the pool and protocol TVL and inputTokenBalance.
+   * @returns Liquidate
+   */
+  liquidate(
```

Comment:
Should we call `countLiquidate` and `countLiquidatee` on the accounts involved from this method? Or do you want to leave that up to the caller?

Reply:
I think it's better that the function handles this automatically. I have made changes to handle this inside the liquidate function, same for other similar functions.